examples/platform/silabs/efr32/rs911x/hal/efx_spi.c

```diff
@@ -141,13 +158,131 @@ void rsi_hal_board_init(void)
     spiTransferLock = xSemaphoreCreateBinaryStatic(&xEfxSpiIntfSemaBuffer);
     xSemaphoreGive(spiTransferLock);
 
+#if defined(EFR32MG24)
+    spi_sem_sync_hdl = xSemaphoreCreateBinaryStatic(&spi_sem_peripheral);
+    xSemaphoreGive(spi_sem_sync_hdl);
+#endif /* EFR32MG24 */
+
     /* GPIO INIT of MG12 & MG24 : Reset, Wakeup, Interrupt */
     sl_wfx_host_gpio_init();
 
     /* Reset of Wifi chip */
     sl_wfx_host_reset_chip();
+    SILABS_LOG("%s completed.", __func__);
+}
+
+#if defined(EFR32MG24)
+
+void SPIDRV_ReInit(uint32_t baudrate)
+{
+    if (EUSART_BaudrateGet(MY_USART) == baudrate)
+    {
+        // EUSART synced to baudrate already
+        return;
+    }
+    EUSART_BaudrateSet(MY_USART, 0, baudrate);
+}
+
+sl_status_t sl_wfx_host_spi_cs_assert(void)
+{
+    configASSERT(spi_sem_sync_hdl);
```

Comment:
Not convinced that the assert should be there at all. With my previous comment, a simple check after the semaphore creation would do the trick